### ISSUE TITLE
Endpoint auth. parameter that is not confidential getting masked in agent logs

### ISSUE THREAD

**Issue by GitHubSriramB** [microsoft/azure-pipelines-agent#1207]
2.122.0

Windows

VSTS

Attaching details about the custom endpoint type & logs that has this issue. 

![endpoint](https://user-images.githubusercontent.com/13654692/30414135-b674ae00-993f-11e7-9b36-7a82b92b336e.png)
![agent-logs](https://user-images.githubusercontent.com/13654692/30414137-ba85e52c-993f-11e7-9c44-8ddd66c42d63.png)

Endpoint details logged in agent:

```json
{
  "data": {},
  "id": "cfa7e53f-5778-423c-af65-507a80224c79",
  "name": "cfa7e53f-5778-423c-af65-507a80224c79",
  "type": "habitatoriginendpoint",
  "url": "https://bldr.habitat.sh",
  "authorization": {
    "parameters": {
      "username": "********",
      "revision": "********",
      "publickey": "********",
      "password": "********",
      "githubauthtoken": "********",
      "useSudo": "********"
    },
    "scheme": "UsernamePassword"
  },
  "isReady": false
}
```

Endpoint contribution type:

```json
{
  "id": "habitat-origin",
  "description": "Habitat Origin",
  "type": "ms.vss-endpoint.service-endpoint-type",
  "targets": ["ms.vss-endpoint.endpoint-types"],
  "properties": {
    "name": "habitatoriginendpoint",
    "displayName": "Habitat Origin",
    "url": {
      "displayName": "Habitat Depot URL",
      "helpText": "URL to the Habitat depot that will be used to deploy to"
    },
    "inputDescriptors": [],
    "authenticationSchemes": [
      {
        "type": "ms.vss-endpoint.endpoint-auth-scheme-basic",
        "inputDescriptors": [
          {
            "id": "username",
            "name": "Origin Name",
            "description": "Name of the Habitat origin",
            "inputMode": "textbox",
            "isConfidential": false,
            "validation": {
              "isRequired": true,
              "dataType": "string"
            }
          },
          {
            "id": "revision",
            "name": "Revision",
            "description": "Revision of the origin to use",
            "inputMode": "textbox",
            "isConfidential": false,
            "validation": {
              "isRequired": true,
              "dataType": "string"
            }
          },
          {
            "id": "publickey",
            "name": "Public Key",
            "description": "Public item of the origin key pair",
            "inputMode": "textarea",
            "isConfidential": false,
            "validation": {
              "isRequired": true,
              "dataType": "string"
            }
          },
          {
            "id": "password",
            "name": "Signing Key",
            "description": "Signing item of the origin key pair",
            "inputMode": "textarea",
            "isConfidential": true,
            "validation": {
              "isRequired": true,
              "dataType": "string"
            }
          },
          {
            "id": "githubauthtoken",
            "name": "GitHub Auth Token",
            "description": "Authentication token for GitHub for publishing Habitat packages.",
            "inputMode": "textbox",
            "isConfidential": "true",
            "validation": {
              "isRequired": true,
              "dataType": "string"
            }
          },
          {
            "id": "useSudo",
            "name": "Use Sudo",
            "description": "Use sudo on habitat commands",
            "inputMode": "combo",
            "isConfidential": false,
            "validation": {
              "isRequired": false,
              "dataType": "string"
            },
            "values": {
              "inputId": "useSudoValues",
              "defaultValue": "1",
              "possibleValues": [
                {
                  "value": "1",
                  "displayValue": "True"
                },
                {
                  "value": "0",
                  "displayValue": "False"
                }
              ]
            }
          }
        ]
      }
    ],
    "helpMarkDown": "Please provide the requested information so that Habitat packages can be built and deployed. The GitHub authentication token requires `user:email` and `read:org` scopes"
  }
}
```

**Comment by bryanmacfarlane** [comment]
Are you sure this is an agent issue (and not lib/service)?

**Comment by GitHubSriramB** [comment]
@bryanmacfarlane - I was referring to this piece of code in agent that needs a fix:
https://github.com/Microsoft/vsts-agent/blob/327cde0be616c9809168b160f9431702b7b46e1c/src/Agent.Worker/Worker.cs#L144

**Comment by GitHubSriramB** [comment]
Fix for this issue is blocked due to a service-side bug in Build. Waiting for the service-side bug to be fixed before we can take up this fix.

@ericsciple - pl. assign this back to us once the service-side bug is fixed.

**Comment by chperich** [comment]
This [feedback ticket](https://developercommunity.visualstudio.com/content/problem/298319/build-logs-have-character-1-replaced-with-asterisk.html) has been raised recently by a user. Character 1 gets masked in the user's logs because of masking all endpoint authorization parameters.

**Comment by peterox** [comment]
Is there any update to this issue? I am also having the problem where all '1' characters are being masked.

**Comment by chperich** [comment]
+ Lova/Omesh, who will be able to address this better.

**Comment by omeshp** [comment]
@ericsciple should be able to provide update on build side issue.

**Comment by ericsciple** [comment]
@peterox I believe this was fixed. Are you using the service "Azure Devops" and the latest agent? Or are you using on-premises TFS? Note you can right click on the pool to update all agents in a pool to the latest version.

**Comment by peterox** [comment]
@ericsciple I've just updated all the agents and I'm still getting all '1''s masked in the logs.

I disabled a Docker task that had an Azure subscription configured and ran the build and the issue was not there. Re-enabling the task and running the job again caused the issue to re-appear.

**Comment by ericsciple** [comment]
@TingluoHuang do you remember context on this? it looks like [this code](https://github.com/Microsoft/azure-pipelines-agent/blob/master/src/Agent.Worker/Worker.cs#L141) is still in master. I thought this was removed months ago?

**Comment by peterox** [comment]
Bump

**Comment by willgarcia** [comment]
Just confirming that the latest Github released version (2.144.2) does not fix this issue after redeployment of my agents.

The HTTP link in the Web UI is also pointing to version 2.144.0 (instead of 2.144.2).

Current version of `worker.cs` on master is from Aug 17, 2018 (`ab40f0b1..`). 

@ericsciple Would you be able to (un)confirm if the bug is still on master or a fixed on another branch?

**Comment by deadlydog** [comment]
We are using hosted Azure DevOps in the cloud with on-prem agents running version `vsts-agent-win-x64-2.144.0` and we are also seeing this problem still.

**Comment by peterox** [comment]
Are we able to get an update on this? It seems there is a fix but it just hasn't been released?

This problem effectively makes the logs in our pipelines useless.

**Comment by willgarcia** [comment]
As spotted by @peterox, removing our Azure DevOps Docker tasks from pipelines seems to solve the problem.

Specific tasks types that require access to secrets may interfere and cause this issue.

**Comment by thewellington** [comment]
We experience this too - and our logs are useless because of it.  How do we get this resolved?

**Comment by Ivers07** [comment]
We are also experiencing this issue. For us, *** replaces all spaces in the logs of any pipeline or release that uses a service connection for our service fabric instance. 

It only seems to apply to tasks within the Job that uses the connection. If the release has more than one job then the other jobs are OK, but not ones that use the service connection.

Also, i found this issue through the feedback ticket mentioned by @chperich 

> This [feedback ticket](https://developercommunity.visualstudio.com/content/problem/298319/build-logs-have-character-1-replaced-with-asterisk.html) has been raised recently by a user. Character 1 gets masked in the user's logs because of masking all endpoint authorization parameters.

but it wasn't very clear from the title or original description that it was the Service Connection causing masking of incorrect characters

**Comment by frackham** [comment]
I'm getting this with strings being set as VSO variables within a powershell script task in Azure Devops (so also via service connection). The strings are not secret, but the *value* matches a secret elsewhere (in this case, a guid... and other guids are not masked). 

I can get around it by masking the value:
![image](https://user-images.githubusercontent.com/2105204/72059720-77a05400-32ca-11ea-9f33-1d4e62d5bed4.png)
_(blanking out some of the values for obvious reasons)_
None of these are secrets, all are set as release (VSO) variables.

1. The first is the value that matches an existing secret.
2. The second has x added at the start and end of the string.
3. The third has x inserted inside the string, unmasking it.

Looks to me like pattern matching is being applied to values against known secrets, to identify whether they should be obfuscated.

In this case, it's fine for me as I'm only using this for debugging, but seems like an odd way to identify secrets.

Agents are up to date, and no difference between agents.

**Comment by grahamauty** [comment]
For me, it's all lowercase 'b's and 'c's that are being masked with ***. These are obviously not secrets! This is happening in every log for every task (40 of them) in the release pipeline, which is running on Azure Devops with the latest agents.

Example:
![image](https://user-images.githubusercontent.com/29207857/84916191-a7dbf200-b0b5-11ea-9b5b-7899079158d9.png)

This makes it very difficult to read most of the logs (the above example is mild compared to some of it). Can someone **please** update the status of this problem which was raised in 2017.

**Comment by Orthaanc** [comment]
In my case it is lower case 'a' being masked.  

At the moment, I'm trying to investigate where the variables that are getting masked are collected.  There are a number of [Pre-defined Pipeline variables](https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml)  that correspond with characters that are being mentioned in this issue.

Build.BinariesDirectory lists the example c:\agent_work\1\**b**
Agent.BuildDirectory (and Pipeline.Workspace) lists /home/vsts/work/**1**
Build.Repository.LocalPath c:\agent_work\1\**s**
Build.ArtifactStagingDirectory c:\agent_work\1\**a**

**Comment by Paxilein** [comment]
For me it's the Azure Region's Name being secret... Which of course makes perfect sense....

**Comment by RamkiUIPath** [comment]
Is there any workaround for this? 
In our pipelines, test results will be published to a URL and runId gets masked 🤦‍♂️

**Comment by ajayt77** [comment]
Any luck fixing masking of the characters issue in logs? I have tried the following
1) Used the latest Ubuntu build agent
2) SET the variable SYSTEM_DONOTMASKMULTILINESECRETS=TRUE

Did not work. Please help

**Comment by StingyJack** [comment]
[This is happening to at least one of our pipelines](https://developercommunity.visualstudio.com/t/numbers-in-pipeline-logs--are-randomly-b/1424076), but its everything from DateTime stamps to task version, etc.   There are no secrets defined in this particular pipeline. 


From a powershell script task
Task versions, timestamps, paths replaced partially with *
```
Starting: PowerShell Script
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : ***.***80.***
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
========================== Starting Command Output ===========================
"C:\Windows\System3***\WindowsPowerShell\v***.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\a\_temp\a60697d3-63d4-4fc5-b0***a-ce7***39aea8d9.ps***'"
[***0***/05/***4 06:***3:39] Transfer summary:
-----------------
Total files transferred: 9
Transfer successfully:   9
Transfer skipped:        0
Transfer failed:         0
Elapsed time:            00.00:00:00
Finishing: PowerShell Script
```

**Comment by StingyJack** [comment]
@jtpetty - not sure if you are the correct person to tag but you have more recent activity than @ericsciple does in this repo. Can someone please address this? Obscured logs make troubleshooting and resolution take longer.

**Comment by itecompro** [comment]
This issue is open for years and no one has yet figured out what's going on!? 🙄

In our pipelines, we are using Variable Groups connected to a Key Vault, right after the Download Secret task, all 1s are replaced with ***.

Have checked all the secrets in the Key Vault. None of them is `1`. Of course, it could be part of a secret like `something_1_something_else`, but not by itself as a whole.

**Comment by anatolybolshakov** [comment]
Hi everyone, prioritizing this item to check if there is something still need to be fixed after the [PR](https://github.com/microsoft/azure-pipelines-agent/pull/3661).

**Comment by StingyJack** [comment]
@anatolybolshakov - thats great. I will direct the tech assigned to the support case we have open to this thread so they can follow up directly if necessary.

**Comment by anatolybolshakov** [comment]
@StingyJack yes correct, we observed the same issue with '1' being masked - when there are task which use Azure service connections.

**Comment by anatolybolshakov** [comment]
Hi @GitHubSriramB closing this one since changes has been rolled out - feel free to ask any other questions.

**Comment by StingyJack** [comment]
@anatolybolshakov - The engineer assigned to a support case we have open for this asked me to check our build logs now, and we are still having the problem (and he is on leave for the next week or so).  Are you sure this is not still awaiting deployment?  It is the presence of an Azure Key Vault task that triggers this for us. 

Its actually kind of bad because some of the things its masking are things I can kind of easily guess, which means I now know the values of some of the secrets. For example
```
2022-03-12T23:40:40.2210221Z ##[section]Starting: Use Node 1***.18.***
2022-03-12T23:40:40.2244542Z ==============================================================================
2022-03-12T23:40:40.2245097Z Task         : Node.js tool installer
2022-03-12T23:40:40.2245714Z Description  : Finds or downloads and caches the specified version spec of Node.js and adds it to the PATH
2022-03-12T23:40:40.2246490Z Version      : 0.***00.0
2022-03-12T23:40:40.2246928Z Author       : Microsoft Corporation
2022-03-12T23:40:40.2247457Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/tool/node-js
2022-03-12T23:40:40.2248059Z ==============================================================================
2022-03-12T23:40:40.3932609Z Downloading: https://nodejs.org/dist/v1***.18.***/node-v1***.18.***-linux-x64.tar.gz
2022-03-12T23:40:40.6562110Z Extracting archive
2022-03-12T23:40:40.6587983Z [command]/usr/bin/tar xC /home/vsts/work/_temp/14775***7***-0fb9-4***8***-bf57-***ae8199eab81 -f /home/vsts/work/_temp/090***8e***c-174b-4ac***-8a76-f95b8***8c88a5
2022-03-12T23:40:41.2737568Z Caching tool: node 1***.18.*** x64
2022-03-12T23:40:41.8463078Z Prepending PATH environment variable with directory: /opt/hostedtoolcache/node/1***.18.***/x64/bin
2022-03-12T23:40:41.9082799Z ##[section]Finishing: Use Node 1***.18.***
```

From this I know that a few of the secret values are 2 and 3 - kind of silly to put a value like this in secrets, but I think its sillier to mask it. Other tasks have similar trivial "secrets" being masked like "dev" or some filesystem paths, etc.. Some build tasks are more revealing, like ones that construct a document like an emailed summary or release notes.  Those tend to have a bunch of *** in the  logs.

**Comment by StingyJack** [comment]
@anatolybolshakov  - Should this fix have made it to ring M201.1 by now?

**Comment by HofmeisterAn** [comment]
@anatolybolshakov We still have the issue with the agent `2.202.1`. How does #3661 cover **not confidential** inputs? It looks like the change does only fix the problem for some `EndpointAuthorizationParameters` parameters, but what about custom extensions (`inputDescriptors`)?

**Comment by alexander-smolyakov** [comment]
Hey @HofmeisterAn, let us take another look at this issue.

**Comment by StingyJack** [comment]
@alexander-smolyakov  also see [this](https://github.com/microsoft/azure-pipelines-agent/issues/1207#issuecomment-1066023954), it's affecting hundreds of pipelines my org has, and we have a support case open for this.

**Comment by anatolybolshakov** [comment]
@StingyJack @HofmeisterAn all service connection parameters are being masked by default for security reasons - there is no generic mechanism at the moment to distinguish between secret/non secret fields of service connection - hence it's not recommended to use fields with short values for custom service connections currently.
Since service connection definitions is the part of ADO itself - I would suggest to open a ticket on https://developercommunity.visualstudio.com/search?space=21 for such possible enhancement

**Comment by HofmeisterAn** [comment]
> there is no generic mechanism at the moment to distinguish between secret/non secret fields

The `inputDescriptor` has a specific field for that: `isConfidential`.

> hence it's not recommended to use fields with short values for custom service connections currently

Where is that?

**Comment by anatolybolshakov** [comment]
> The `inputDescriptor` has a specific field for that: `isConfidential`.

This is the schema of custom endpoint, but agent relies on `ServiceEndpoint` definition while working with service connections data (coming from external assembly - containing some generic interfaces and classes for interaction with Azure DevOps). This assembly is not the part of the agent, but is more related to Azure DevOps side - updating this would probably require other changes on ADO side as well I believe...

> Where is that?

This is actual for inputs specified in custom endpoint schema (with input descriptors).

**Comment by StingyJack** [comment]
> all service connection parameters are being masked by default for security reasons - there is no generic mechanism at the moment to distinguish between secret/non secret fields of service connection - hence it's not recommended to use fields with short values for custom service connections currently.

Sorry, @anatolybolshakov  - I didnt say _anything_ about service connections. We use service connections for pipelines because my org's repos are hosted on GitHub, and those dont cause this. The inappropriate masking of values in the logs is affecting pipelines that have an `Azure Key Vault` task that is enabled. If the task is disabled this problem goes away. 

If you think that this is just a problem with endpoint authorization it means the actual problem is still not being recognized.  The pipelines are masking values parts of the logs that will not contain or display secrets, and that behavior is allowing me to guess what some of the secret values are.  **The masking logic is actually exposing the secrets that it is supposed to be hiding.**  

One of the responses we got on our support case was that "this was by design". I've gently rejected that and am giving them the benefit of the doubt by choosing to believe that whomever said that also just doesnt quite understand what is being explained or reported as the problem. Also because there is no documentation that exists that warns users of the potential exposure, which I would expect of any potential leakage that could be caused by specific configurations or misuse.

**Comment by StingyJack** [comment]
I would also like to point out that all of the reports of this issue on the developer community site are getting closed as duplicates and are referring to this issue as the place where we should be tracking, so please do not ask us to "open another issue" or anything like that because we have reported the issue correctly already.

**Comment by HofmeisterAn** [comment]
> The inappropriate masking of values in the logs is affecting pipelines that have an Azure Key Vault task that is enabled.

It also happens for custom extensions.

**Comment by StingyJack** [comment]
@anatolybolshakov - We only have a few builds that require us to host an agent. The rest are all AzDo hosted agents which are running the pipelines that have the inappropriate masking, so I'm only looking into these hosted agents. From a recent log...
```
2022-05-11T00:30:48.6258170Z Current agent version: '2.202.1'
```
I ran a pipeline with debug = true and System Diagnostics enabled and attached the logs to the support case we have open. I believe the support engineer has forwarded them to you at this point. 

> We also investigated task itself - it masks only one secret value, but we determined that it can't be '1'.

The logs attached appear to have values like "1", "2", "Dev", etc. being masked.  Its masking them in places like the task version header, or in the timestamps, etc. See one of the examples I have already shared for a better idea.

**Comment by KonstantinTyukalov** [comment]
Hey everyone!
We are introducing new agent knob: `AGENT_IN_SECRET_LENGTH` which will let you not to mask short secrets.
You can specify values from `0` (all secrets will be masked as usual) to `4` (secrets shorten than 4 symbols will be ignored)
Also, negative values will mean masking all the secrets as usual.

example:
```yml
  variables:
    AGENT_MIN_SECRET_LENGTH: 1
```

also it's available as environment variable.

This feature will be available in the next agent release, we will notify you once it will be rolled out.
Thanks!

**Comment by DenisRumyantsev** [comment]
@StingyJack I have tested that timestamps remain and are not affected by secret masking:

![image](https://user-images.githubusercontent.com/61472718/198677117-bd14f730-d0b8-49e7-852d-0c9058b7595b.png)

Also, please notice that the name of the variable is `AGENT_MIN_SECRET_LENGTH` , not `AGENT_IN_SECRET_LENGTH`

**Comment by StingyJack** [comment]
> I have tested that timestamps remain and are not affected by secret masking

Yes and I notice the log is now totally useless as well. 



> Also, please notice that the name of the variable is `AGENT_MIN_SECRET_LENGTH` , not `AGENT_IN_SECRET_LENGTH`

That makes more sense but its not what @KonstantinTyukalov said it was called. 

> We are introducing new agent knob: AGENT_IN_SECRET_LENGTH which will let you not to mask short secrets.

Which is not as important as that you are still missing the point. We dont want you to avoid masking secrets.  This solution makes the problem worse as it allows me to expose secrets directly instead of just inferring them indirectly.

We want you to avoid masking data that is not a secret.

**Comment by StingyJack** [comment]
@DenisRumyantsev @KonstantinTyukalov @anatolybolshakov @alexander-smolyakov  - I pointed out that your proposed solution creates an even larger problem than was being reported, and you have nothing to say in the last month?  

I really hope that you are not still considering this AGENT_SECRET_MIN_LENGTH as viable, because nobody asked for you for a knob that let us expose any secrets. We asked you to stop making the secret values easy to guess.

**Comment by at-besa** [comment]
> This issue is open for years and no one has yet figured out what's going on!? 🙄
> 
> In our pipelines, we are using Variable Groups connected to a Key Vault, right after the Download Secret task, all 1s are replaced with ***.
> 
> Have checked all the secrets in the Key Vault. None of them is `1`. Of course, it could be part of a secret like `something_1_something_else`, but not by itself as a whole._
_____________________________________________________________________________

> Hey everyone! We are introducing new agent knob: `AGENT_IN_SECRET_LENGTH` which will let you not to mask short secrets. You can specify values from `0` (all secrets will be masked as usual) to `4` (secrets shorten than 4 symbols will be ignored) Also, negative values will mean masking all the secrets as usual._
_____________________________________________________________________________

The knob is not working for the bug that '1' is masked, we also checked our secrets and there is none of them having a 1. Also interesting fact is that it appears to happen only on the first stage, if the pipeline does have more stages the 1's are not masked. The pipelines use the same agent for these stages, and the agent version is 2.217.2, which is the newest as by the time writing this.

**So result is neither is the bug with suddenly masking 1 solved nor is there any workaround or solution to not have guessable logs where secrets are masked out of text which shouldn't be masked.**

@KonstantinTyukalov @anatolybolshakov

**Comment by at-besa** [comment]
> > This issue is open for years and no one has yet figured out what's going on!? 🙄
> > In our pipelines, we are using Variable Groups connected to a Key Vault, right after the Download Secret task, all 1s are replaced with ***.
> > Have checked all the secrets in the Key Vault. None of them is `1`. Of course, it could be part of a secret like `something_1_something_else`, but not by itself as a whole._
> 
> > Hey everyone! We are introducing new agent knob: `AGENT_IN_SECRET_LENGTH` which will let you not to mask short secrets. You can specify values from `0` (all secrets will be masked as usual) to `4` (secrets shorten than 4 symbols will be ignored) Also, negative values will mean masking all the secrets as usual._
> 
> The knob is not working for the bug where that '1' is masked, we also checked our secrets and there is none of them having a 1. Also interesting fact is that it appears to happen only on the first stage, if the pipeline does have more stages the 1's are not masked except the first stage. The pipelines use the same agent for these stages, and the agent version is 2.217.2, which is the newest as by the time writing this.
> 
> **So result is neither is the bug with suddenly masking 1 solved nor is there any workaround or solution to not have guessable logs where secrets are masked out of text which shouldn't be masked.**
> 
> @KonstantinTyukalov @anatolybolshakov

This also happens with the new Azure Agent with Version 3.217.1 as we just tried the pre-release version.

**Comment by KonstantinTyukalov** [comment]
@at-besa Do you have this or other extension in your pipeline? https://github.com/microsoft/azure-pipelines-agent/issues/1207#issue-257604223

Also, have you tried `AZP_IGNORE_SECRETS_SHORTER_THAN` knob name? Note that `AGENT_MIN_SECRET_LENGTH` knob name is not valid for now.

> Also interesting fact is that it appears to happen only on the first stage, if the pipeline does have more stages the 1's are not masked except the first stage.

What tasks do you have in this first stage?

**Comment by StingyJack** [comment]
> Also, have you tried AZP_IGNORE_SECRETS_SHORTER_THAN knob name? Note that AGENT_MIN_SECRET_LENGTH knob name is not valid for now.

@KonstantinTyukalov - this knob does not fix the problem that was reported. It actually makes the situation worse, thus invalidating it's intended purpose. Several users have told you this now. Please revert the knob and come up with a proper fix.

**Comment by at-besa** [comment]
> @at-besa Do you have this or other extension in your pipeline? [#1207 (comment)](https://github.com/microsoft/azure-pipelines-agent/issues/1207#issue-257604223)
> 
> Also, have you tried `AZP_IGNORE_SECRETS_SHORTER_THAN` knob name? Note that `AGENT_MIN_SECRET_LENGTH` knob name is not valid for now.
> 
> > Also interesting fact is that it appears to happen only on the first stage, if the pipeline does have more stages the 1's are not masked except the first stage.
> 
> What tasks do you have in this first stage?

Yes i have tried the knob, the 1's are not masked anymore when i turn the knob value of AZP_IGNORE_SECRETS_SHORTER_THAN  to 1, although this does not make any sense, as mentioned there is **no secret** which does even have this value. I agree with @StingyJack that this is just an workaround and not the solution to the actual problem.

regarding the tasks used in the first stage:
          - checkout: self
          - task: Bash@3
          - task: Docker@2 

The whole pipeline is running on an azure hosted vm with rocky as operating system. 

as there is no secret stuff in this section i can share the yaml of this whole stage:

```
  - stage: Build
    jobs:
      - job: Build
        timeoutInMinutes: 10
        cancelTimeoutInMinutes: 2
        steps:
          - checkout: self
            fetchDepth: 2
            clean: true
            submodules: recursive
             
          ############ convert repo name to lowercase for openshift  ############
          - task: Bash@3
            displayName: 'Convert Repo to lowercase'
            inputs:
              targetType: 'inline'
              script: |
                LowerRepoName=$(echo $(Build.Repository.Name) | tr  [:upper:] [:lower:])
                echo "##vso[task.setvariable variable=RepoName]$LowerRepoName"
          
          
          ############ build image, tag and push it to registry  ############
          - task: Docker@2
            displayName: 'docker build and push'
            inputs:
              containerRegistry: 'openshift-registry'
              repository: '$(imageName)'
              command: 'buildAndPush'
              Dockerfile: '**/Dockerfile'
              tags: |
                $(Build.BuildNumber)
                latest
```

**Comment by at-besa** [comment]
any news on that ?

**Comment by at-besa** [comment]
180 days and still no comment of devs

**Comment by StingyJack** [comment]
Turn off this bot. It just makes the maintainers come across as "rude" instead of just "busy"

**Comment by StarWars999123** [comment]
> Turn off this bot. It just makes the maintainers come across as "rude" instead of just "busy"

would consider them extremly ignorant now... This issue annoys me almost every day.

**Comment by roospa** [comment]
We have been monitoring this issue for a quite some while now...amazing....still today the issue is not fixed and logs are becoming fairly unusable as much of non secret information is being masked.
`2024-11-06T13:54:19.5517066Z Current agent version: '3.242.1' `

**Comment by ThePhilippO** [comment]
In the latest version of the agent, this error still occurs.
